R/tm_outliers.R

```diff
@@ -246,52 +247,56 @@ srv_outliers <- function(id, datasets, reporter, outlier_var,
       }
     })
 
-    merged_data <- teal.transform::data_merge_srv(
+    anl_merged_input <- teal.transform::merge_expression_srv(
       selector_list = reactive_select_input,
-      datasets = datasets,
+      datasets = data,
+      join_keys = attr(data, "join_keys"),
       merge_function = "dplyr::inner_join"
     )
 
-    is_cat_filter_spec <- inherits(categorical_var[[1]]$filter[[1]], "filter_spec")
-    cat_dataname <- categorical_var[[1]]$dataname
-
-    common_code_chunks <- reactive({
-      validate(need(!is.null(reactive_select_input()$outlier()), "Please select an outlier variable dataset"))
+    anl_merged_q <- reactive({
+      teal.code::new_quosure(env = data) %>%
+        teal.code::eval_code(as.expression(anl_merged_input()$expr))
+    })
 
-      # Create a private stack for this function only.
-      common_stack <- teal.code::chunks$new()
+    merged <- list(
+      anl_input_r = anl_merged_input,
+      anl_q_r = anl_merged_q
+    )
 
-      common_stack_push <- function(...) {
-        teal.code::chunks_push(..., chunks = common_stack)
-      }
+    is_cat_filter_spec <- inherits(categorical_var[[1]]$filter[[1]], "filter_spec")
+    cat_dataname <- categorical_var[[1]]$dataname
 
+    common_code_q <- reactive({
       input_catvar <- input[[extract_input(
         "categorical_var",
         cat_dataname,
         filter = is_cat_filter_spec
       )]]
 
-      teal.code::chunks_push_data_merge(merged_data(), common_stack)
-      outlier_var <- as.vector(merged_data()$columns_source$outlier_var)
-      categorical_var <- as.vector(merged_data()$columns_source$categorical_var)
+      ANL <- merged$anl_q_r()[["ANL"]]
+      quosure <- merged$anl_q_r()
+
+      outlier_var <- as.vector(merged$anl_input_r()$columns_source$outlier_var)
+      categorical_var <- as.vector(merged$anl_input_r()$columns_source$categorical_var)
       order_by_outlier <- input$order_by_outlier # nolint
       method <- input$method
       split_outliers <- input$split_outliers
 
       validate(need(outlier_var, "Please select a variable"))
-      teal.code::chunks_validate_custom(
-        substitute(expr = length(unique(ANL[[outlier_var]])) > 1, env = list(outlier_var = outlier_var)),
-        msg = "Variable has no variation, i.e. only one unique value",
-        chunks = common_stack
-      )
+
+      code <- substitute(expr = length(unique(ANL[[outlier_var]])) > 1, env = list(outlier_var = outlier_var))
+      msg <- "Variable has no variation, i.e. only one unique value"
+      validate(need(all(eval(code, quosure@env)), msg))
```

Comment:
Is it necessary to evaluate `code` here in the context of `quosure@env`?

Reply:
@Polkas @mhallal1 I'll fix this issues related to usage of @env